internal/keepers/service.go

```diff
@@ -219,7 +136,7 @@ func (s *onDemandUpkeepService) parallelCheck(ctx context.Context, keys []types.
 					success++
 					// cache results
 					s.cache.Set(string(result.Data.Key), result.Data, defaultExpiration)
```

Comment:
not sure if easy, but would be great if we can just reuse the checkUpkeep function and not worry about cache in this function

Reply:
So the workers just call the service's CheckUpkeep? Makes sense. I really want to clean that section up anyway. It'll be a good todo item for later.